.travis.yml

```diff
@@ -1,43 +1,117 @@
 language: rust
 sudo: false
 
-# We aim to test all the following in any combination:
-# - standard tests, benches, documentation, all available features
-# - pinned stable, latest stable, beta and nightly Rust releases
-# - Linux, OS X, Android, iOS, bare metal (i.e. no_std)
-# - x86_64, ARMv7, a Big-Endian arch (MIPS)
+# We support too many combinations of Rust releases, crate features, operating
+# systems, and architectures to even remotely test all combinations.
+# Yet it turns out we can test most of these independent of each other, because
+# they serve different goals or test different pieces of code.
+#
+# RUST RELEASES
+# Goal: make sure we don't use language features unavailable on a certain
+#       version, and build without warnings.
+# We have different builders use 4 Rust releases, a pinned stable release,
+# the latest stable, beta and nightly.
+#
+# ARCHITECTURES
+# Goal: test against issues caused by differences in endianness, pointer sizes,
+#       etc.
+# We run tests on 4 different architectures.
+# - x64_84, default on Travis (Linux) and AppVeyor (Windows)
+# - i686, second AppVeyor (Windows) configuration
+# - MIPS, big-endian Linux emulated with QEMU (thanks to Trust)
+# - ARMv7, Android emulated with QEMU (thanks to Trust)
+#
+# OPERATING SYSTEMS
+# Goal: test on many operating systems, to verify the OsRng code, which is
+#       mostly architecture-independend.
+# We run tests on Linux, OS X, Windows, Android (emulated), and Node.js (using
+# cargo-web).
+# One builder cross-compiles for many of the remaining OSes, which ensures we
+# keep building, but doesn't run tests.
+# OSes supported by Rand but which we can't cross-compile because there
+# is no pre-build standard library available: Dragonfly BSD, Haiku, OpenBSD.
+#
+# CRATE FEATURES, TESTS, AND SUB-CRATES
+# Goal: Run unit tests, doctests, examples, and benchmarks (converted to tests)
+#       for all crates, in configurations that cover all interesting
+#       combinations of features.
+# Tests run on rand:
+# - test no_std support, but only the unit tests:
+#   `cargo test --lib --no-default-features`
```

Comment:
Do we want --lib or --tests?

Reply:
On master we have this comment:
```yaml
        # TODO: use --tests instead of --lib on more recent compiler
```

Reply:
They do exactly the same (because we are only a library), but `--lib` is available in Rust 1.22.0, `--tests` isn't.